Working install v0.1

SiM22 opened the pull request:
I know there is already a very similar PR but below I will list changes and features added.  Have renamed all the comments/directories that mention `paperless-ng` to `paperless-nix`

* Renamed all the comments/directories that mention `paperless-ng` to `paperless-ngx` [Note: Variables are still not updated to ngx
* Added paperless export directory
* Added NFS mount possibilities for external consumption directory [if using an LXC container please make the relevant changes to allow the LXC to use NFS]
* Added Tika/Gotenburg endpoint variable
* Added step to upgrade pip within virtualenv prior to running the install of requirements [This is because if the step to upgrade]

Currently working with v1.8.0

SiM22 commented:
@shamoon @qcasey any chance someone can take a look? 

I do not mind being an active member on the ansible repo if you need more eyes on this side of the fence

shamoon commented:
Appreciate the PR. The reason we separated out the ansible stuff was because we don’t have a lot of expertise on the dev team with it. For myself for example I am of almost no use for this. If you’d like to become the go-to person for this that might make sense, qcasey I believe there really isn’t anyone else working on this at the moment correct?

stumpylog commented:
I actually use Ansible to configure and maintain my own systems, though I don't deploy paperless with it (I dockerize everything possible).  So I'll have a look through.

qcasey commented:
> qcasey I believe there really isn’t anyone else working on this at the moment correct?

No one is maintaining this as far as I know. Just #2 which this claims to be an extension of anyway.

stumpylog if this tests OK and @SiM22 wants to maintain I think that would be a good solution.

stumpylog commented on tasks/main.yml:
```diff
@@ -254,18 +252,27 @@
     removes: "{{ paperlessng_directory }}/paperless.conf"
 
-- name: configure paperless-ng
+- name: configure paperless-ngx
```
This fails with the following vars undefined:
- paperlessng_url
- paperlessng_enable_update_check

SiM22 replied:
Added both into defaults, missed out initially as was part of my group_vars.

stumpylog commented on tasks/main.yml:
```diff
@@ -105,79 +123,58 @@
 
 - block:
-    - name: backup current paperless-ng installation
+    - name: backup current paperless-ngx installation
```
If `/opt/paperless-ngx` doesn't already exist, this step fails.

SiM22 replied:
The flags were wrong on the `register current state` step. I was testing the upgrade to 1.8.0. Both flags now set to follow the fresh installation path

stumpylog commented on tasks/main.yml:
```diff
@@ -254,18 +252,27 @@
-- name: configure paperless-ng
+- name: configure paperless-ngx
   lineinfile:
     path: "{{ paperlessng_directory }}/paperless.conf.template"
```
This file doesn't exist.  It appears to be skipping any installation if nothing exists already.

SiM22 replied:
Same as above. The flags were wrong on the register current state step. I was testing the upgrade to 1.8.0. Both flags now set to follow the fresh installation path

stumpylog reviewed (changes requested):
I tried to give this branch a run in a VM.  Although it does get somewhat into the setup, I'm not able to run it to completion just yet.

qcasey commented on README.md:
```diff
@@ -1,92 +1,92 @@
 ### :warning: THIS REPO IS UNTESTED! Ansible is not currently supported by paperless-ngx. Use only at your own peril. :warning:
```
Agreed, the warning can be removed.

SiM22 replied:
Removed warning

SiM22 commented:
ok let me get some of those things fixed, this is probably related to me using a copy paste technique from the one i got running

SiM22 commented on README.md:
```diff
@@ -1,92 +1,92 @@
+  roles:
+  - role: paperless-ngx
+    become: yes
+    Role Variables
```
Fixed

SiM22 commented on README.md:
````diff
@@ -1,92 +1,92 @@
+```paperlessng_version: latest```
+The [release](https://github.com/paperless-ngx/paperless-ngx/releases) archive version of paperless-ngx to install.
+`latest` stands for the latest release of paperless-ngx.
+To install a specific version of paperless-ngx, use the tag name of the release, e. g. `1.4.4` when the tag is `ngx-1.4.4`.
````
default set to 1.8.0

SiM22 commented on README.md:
```diff
@@ -1,92 +1,92 @@
-    paperlessng_redis_host: localhost
-    paperlessng_redis_port: 6379
 
+Additional variables available in this role are listed below, along with default values:
```
added as a table. probably could be prettier

SiM22 commented on tasks/main.yml:
```diff
@@ -511,20 +533,81 @@
+      - nfs-utils
+      - nfs4-acl-tools
+    state: present
+  when: ansible_os_family == "RedHat" and paperlessng_consumption_dir_nfs_enabled and paperlessng_is_lxc != "True"
```
added as list

SiM22 commented on tasks/main.yml:
```diff
@@ -4,11 +4,26 @@
+  when: item.mount == '{{ paperlessng_consumption_dir }}'
+  loop: "{{ ansible_facts.mounts }}"
+
+- name: Show if consumption mount exists
```
removed as no idea why i had it added previously. May just be a case of working through errors and putting stuff in. But not required to get role to deploy

SiM22 commented on tasks/main.yml:
```diff
@@ -105,79 +123,58 @@
-  when: not(paperlessng_version | regex_search("^(ng-)?(\d+\.\d+\.\d+)$"))
-
-- name: check for paperless-ng installation
+        paperlessng_commit: "{{ commit.json | json_query('sha') }}"
```
Added as a dependency in the Readme

SiM22 commented on README.md:
````diff
@@ -1,92 +1,92 @@
+Directory used for the virtual environment for paperless-ngx.
+```
+paperlessng_ocr_languages:
+- eng
````
This is now part of the table

SiM22 commented:
ok, I believe I have addressed all the comments. Most of them were due to me getting the role to work and then adding to it. A fresh set of eyes has helped get it a bit more tidy :-)

stumpylog reviewed (approved):
I made some changes directly, with which I was able to reach the end of my playbook and re-run it without a failure.  

There's more which can be cleaned up in the future if people want to tackle it, but this works for me as is.
